src/codons.py

```diff
@@ -19,6 +19,11 @@
 
 
 def split_codons(dna: str) -> list[str] | None:
+    if len(split_codons) % 3 != 0:
+        return None
+    if len(split_codons) % 3 == 0:
```

Comment:
If you return in an `if` branch, you don't need an `else` part (or an `if` where you test for the complement of the first one). You know that you would have left the function if you went into the first branch, so if you are past the `if`-block you can just return the expression you want.